### ISSUE TITLE
[Task]: DSL call slower than other API calls.

### ISSUE THREAD

**Issue by SatishGandham** [appsmithorg/appsmith#23637]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

In load testing it was found that the DSL call is slower than other API calls, when it just have to get a single record from the DB.

Identify the reason for the slowness and fix it.

**Comment by somangshu** [comment]
@SatishGandham please also write the next steps that need to be taken here.

cc @NilanshBansal @ginilpg

**Comment by SatishGandham** [comment]
- List down what the API does, for a layman it should just get a record by ID.
- Is the response augmented by details of response from other queries? Is it required? Is it parallel or sequential?  
- What extra processing does it do on the data, is it required? Is it slowing the API? Can we optimize it? By how much? How does it compare to overall API time.
- Check the query plan.

We should do proper performance testing before pushing the changes.

**Comment by NilanshBansal** [comment]
Getting started on this today. @somangshu @SatishGandham @ginilpg

**Comment by NilanshBansal** [comment]
[Here](https://www.notion.so/GET-DSL-API-Performance-Testing-b4a9b02483fa47c19f01503b67a42923) are the results for functional analysis of the API. We have found out that the maximum time taken in this API is by the findById function in our backend code, which does a database query to fetch the page by Id in the database.
We found out that the time taken to query records is somewhat proportional to the size of the record from the mongodb (have verified this using a [python](https://gist.github.com/NilanshBansal/8b61b5f89e5ef26e3571b7d39acf829a) script as well).
If we want to optimise this API further, we need to optimise the internal mongo query or we update the benchmarking nos for this API. 
In this regard, we have started an email thread with Archit from MongoDB, with Somangshu and Shri in CC.

We are waiting on their reply. From a java code POV, there's not much more to what we have already done here. 
CC: @somangshu

**Comment by NilanshBansal** [comment]
Closing this with ref to the above explanation.

**Comment by somangshu** [comment]
leaving this issue open as we still need work to be done here. 
cc @shubham7saxena7 @rajatagrawal
